internal/core/legacy/services/audit/audit.go

```diff
@@ -113,7 +115,113 @@ func (a *Audit) BatchCreateAudit(reqs []apistructs.Audit) error {
 
 // List 通过参数过滤事件
 func (a *Audit) List(param *apistructs.AuditsListRequest) (int, []model.Audit, error) {
-	return a.db.GetAuditsByParam(param)
+	reqParam := &model.ListAuditParam{
+		StartAt:      param.StartAt,
+		EndAt:        param.EndAt,
+		FDPProjectID: param.FDPProjectID,
+		UserID:       param.UserID,
+		TemplateName: param.TemplateName,
+		PageNo:       param.PageNo,
+		PageSize:     param.PageSize,
+		ClientIp:     param.ClientIP,
+		ScopeType:    param.ScopeType,
+	}
+
+	if param.Sys {
+		// if scope_type is sys，there is no need to perform parameter validation
+		reqParam.ProjectID = param.ProjectID
+		reqParam.AppID = param.AppID
+		reqParam.OrgId = param.OrgID
+	} else {
```

Comment:
Do not use else here. You can return inside sys code branch directly.
And you can extract non-sys scope logic to a function, and add unit-test for it.

Reply:
has been fixed